lib/wallet/src/Cardano/Wallet/Transaction.hs

```diff
@@ -357,6 +364,71 @@ data DelegationAction
 instance Buildable DelegationAction where
     build = genericF
 
+data DepositEvent = DepositTaken !Coin | DepositReturned !Coin
+    deriving (Eq, Generic, Show)
+
+data DepositEventParams = DepositEventParams
+    { protocolParameterDeposit :: !Coin
+    , txResolvedInputs :: ![Coin]
+    , txResolvedCollateralInputs :: ![Coin]
+    , txOutputs :: ![Coin]
+    , txCollateralOutputs :: ![Coin]
+    , txFee :: !Coin
+    , txWithdrawals :: ![Coin]
+    } deriving (Eq, Generic, Show)
+
+-- Given information about a particular transaction (represented by
+-- the 'DepositEventParams'), determine whether a deposit was taken or
+-- a refund was issued.
+--
+-- Caveat #1: it is possible (though unlikely) that transaction registers or
```

Comment:
it is possible to construct such a cbor, but not possible to do it through construct transaction. We have at most one delegation action possible now.